apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -90,6 +100,43 @@ public void StartSpan(string name, StartSpanOptions options = null)
                 }
 
                 _traceStack.Push(span);
+                AnnotateSpan(options.Labels);
+            }
+
+            return new Span(this);
+        }
+
+        /// <inheritdoc />
+        public void RunInSpan(Action action, string name, StartSpanOptions options = null)
+        {
+            using (StartSpan(name, options))
+            {
+                try
+                {
+                    action();
+                }
+                catch (Exception e)
```

Comment:
It may be better to use an exception filter here:

```
catch (Exception e) when (SetStackTraceAndReturnFalse(e))
{
}

private bool SetStackTraceAndReturnFalse(Exception e)
{
    SetStackTrace(new StackTrace(e, true));
    return false;
}
```

Definitely test carefully before making this change, because that changes the timing of when the stack trace will be acquired, which could change things significantly... but it does make the exception handling more transparent in terms of final stack.